### PR TITLE
Answer:11

### PR DESCRIPTION
**Pull request by tom-englert**
## Checklist for challenge submission

- [x] Start your PR message with Answer:${challenge_number}

### CODE REVIEW

File: apps/rxjs-pipe-bug/src/app/app.service.ts [hunk 1]
```diff
@@ -8,15 +8,13 @@ export class AppService {
...
       take(1),
-      mergeMap((topicToDelete) =>
-        topicToDelete.length > 0
-          ? topicToDelete
-              .map((t) => this.dbService.deleteOneTopic(t.id))
-              .reduce((acc, curr) => merge(acc, curr), of(true))
-          : of(true)
+      switchMap(topicsToDelete =>
+        forkJoin(topicsToDelete.map((t) => this.dbService.deleteOneTopic(t.id))).pipe(
+          map(results => !results.some(result => !result)))
```
Comment (tomalaforge) [inline apps/rxjs-pipe-bug/src/app/app.service.ts:17]: you can also use the `every` operator to avoid having a double negative condition

**Comment by tomalaforge** [comment]
nice work

File: apps/rxjs-pipe-bug/src/app/app.service.ts (hunk 1 above)
Comment (tom-englert) [reply apps/rxjs-pipe-bug/src/app/app.service.ts:17]: yes, meanwhile I found the docs that `every `on an empty array works as needed here.

File: apps/rxjs-pipe-bug/src/app/app.service.ts [hunk 2]
```diff
@@ -8,15 +8,13 @@ export class AppService {
...
       take(1),
-      mergeMap((topicToDelete) =>
-        topicToDelete.length > 0
-          ? topicToDelete
-              .map((t) => this.dbService.deleteOneTopic(t.id))
-              .reduce((acc, curr) => merge(acc, curr), of(true))
-          : of(true)
+      switchMap(topicsToDelete =>
+        forkJoin(topicsToDelete.map((t) => this.dbService.deleteOneTopic(t.id))).pipe(
+          map(results => results.every(result => result)))
```
Comment (tomalaforge) [inline apps/rxjs-pipe-bug/src/app/app.service.ts:17]: 👍

**Pull request closed without merging**